go/src/pluginmgr/pluginmgr_common/pluginmgr_helper.go

```diff
@@ -0,0 +1,740 @@
+	return nil
+}
+
+// To-Do : Goutham : Dynamic plugins are not handled in V1 cut. Add this code later
+/*
+ * Calls plugin init() and f its successful calls registerAction() to engine. This call may block.
+ *
+ * Input:
+ *  pluginname - plugin name
+ *  pluginVersion - plugin version
+ *  isDynamic - true if plugin is dynamic, false if plugin is static
+ *
+ * Output:
+ *  none -
+ *
+ * Return:
+ *  error - error message or nil on success
+ */
+func (plmgr *PluginManager) AddPlugin(pluginName string, pluginVersion string, isDynamic bool) error {
+	// 1.Check if plugin is already loaded
+	plmgr.mu.Lock()
+	if _, ok := plmgr.plugins[pluginName]; ok {
+		return lomcommon.LogError("plugin with name %s and version %s is already loaded", pluginName, pluginVersion)
+	}
+	plmgr.mu.Unlock()
+
+	// 2.Get plugin specific details from actions config file and add any additional info to pass to plugin's init() call
+	var pluginData plugins_common.PluginData
+	if actionCfg, ok := configMgr.ActionsConfig[pluginName]; ok {
+		pluginData.Timeout = actionCfg.Timeout
+		pluginData.HeartbeatInt = actionCfg.HeartbeatInt
+		pluginData.ActionKnobs = actionCfg.ActionKnobs
+	} else {
+		return lomcommon.LogError("plugin %s not found in actions config file", pluginName)
+	}
+
+	// 3.Check if plugin disabled flag is set or not in the actions config file.
+	if configMgr.ActionsConfig[pluginName].Disable {
+		return lomcommon.LogError("plugin %s is disabled", pluginName)
+	}
+
+	// 4.Create new plugin instance
+	pluginID := plugins_common.PluginId{Name: pluginName, Version: pluginVersion}
+	plugin, err := CreatePluginInstance(pluginID, pluginData) // returns Plugin interface pointing to new plugin struct
+	if err != nil {
+		return err
+	}
+
+	// 5.Check if plugin name and version from proc_conf.json file matches the values in static plugin. If not log periodic log
+	if id := plugin.GetPluginID(); id.Name != pluginName || id.Version != pluginVersion {
+		plmgr.AddPeriodicLogWithTimeouts("Plg_mgr_AddPlugin_"+lomcommon.GetUUID()+"_"+pluginName,
+			fmt.Sprintf("Plugin ID does not match provided arguments: got %s %s, expected %s %s", id.Name, id.Version, pluginName, pluginVersion),
+			PLUGIN_INIT_PERIODIC_TIMEOUT_DEFAULT, PLUGIN_INIT_PERIODIC_FALLBACK_TIMEOUT_DEFAULT, "")
+
+		return errors.New("")
+	}
+
+	// 6.Add plugin to plugin manager
+	plmgr.mu.Lock()
+	plmgr.plugins[pluginName] = plugin
+	plmgr.plugins[pluginName].SetPluginStage(plugins_common.PluginStageLoadingSuccess)
+	plmgr.mu.Unlock()
+
+	// 7.periodic timer unique string
+	uid := "Plg_mgr_AddPlugin_" + lomcommon.GetUUID() + "_" + pluginName
+
+	// 8.timer for plugin's init() call errors - SHort timeout - Default 5min
+	logTimer := time.NewTimer(PLUGIN_INIT_PERIODIC_TIMEOUT_DEFAULT)
+	if logTimer == nil {
+		delete(plmgr.plugins, pluginName) // Verify : Goutham : Is this needed. Can't we store failed plugins ?
+		return lomcommon.LogError("Failed to create logTimer")
+	}
+	if !logTimer.Stop() {
+		<-logTimer.C
```

Comment:
2 timers. 
logTimer -- If init() return error, then only this timer starts. Default is 5min. So every 5min, it will log error untill 1 hour. Then timer is reset to 1Hour. So every 1Hour, it logs error message 
initTimeout - This timer is for init() call timeout. Default i 60 sec. We expect init() to return in 60 sec, if not then again periodic error message is logged for 5min ....then 1Hour untill init() returns some value. In that case, then timer is stopped and loging is also stopped too.

Reply:
So here, intention is to stop logTimer timer and start only when there is error from init(). So thats why stop() is called. According to https://pkg.go.dev/time#NewTimer, when anytime stop() is called, its best to drain timer channel C.

Reply:
Not needed after simplified logic.